lib/telemetry_metrics.ex

```diff
@@ -317,36 +353,40 @@ defmodule Telemetry.Metrics do
 
   # Helpers
 
-  @spec validate_event_or_metric_name!(term()) ::
-          normalized_metric_name() | :telemetry.event_name() | no_return()
-  defp validate_event_or_metric_name!(list) when is_list(list) do
-    if Enum.all?(list, &is_atom/1) do
-      list
+  @spec validate_metric_or_event_name!(term()) :: [atom(), ...]
+  defp validate_metric_or_event_name!(metric_or_event_name) when metric_or_event_name == [] or metric_or_event_name == "" do
+    raise ArgumentError, "metric or event name can't be empty"
+  end
+
+  defp validate_metric_or_event_name!(metric_or_event_name) when is_list(metric_or_event_name) do
+    if Enum.all?(metric_or_event_name, &is_atom/1) do
+      metric_or_event_name
     else
       raise ArgumentError,
-            "expected event or metric name to be a list of atoms or a string, " <>
-              "got #{inspect(list)}"
+            "expected metric or event name to be a list of atoms or a string, got #{
+              inspect(metric_or_event_name)
+            }"
     end
   end
 
-  defp validate_event_or_metric_name!(event_or_metric_name)
-       when is_binary(event_or_metric_name) do
-    segments = String.split(event_or_metric_name, ".")
+  defp validate_metric_or_event_name!(metric_or_event_name)
+       when is_binary(metric_or_event_name) do
+    segments = String.split(metric_or_event_name, ".")
 
     if Enum.any?(segments, &(&1 == "")) do
       Logger.warn(fn ->
```

Comment:
I know this is not related to this PR, but I think we should raise here. It is a bad argument for all purposes, then we can remove the `Enum.filter(fn s -> s != "" end)` call below.